beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs

```diff
@@ -64,7 +64,6 @@ impl PeerSyncStatus {
 
     pub fn update(&mut self, new_state: PeerSyncStatus) -> bool {
         if *self == new_state {
-            *self = new_state;
```

Comment:
the problem here is that `PeerSyncStatus` uses an == that resembles a `matches!(..)`. So removing this line does change the logic

Reply:
Good catch! I was wondering if Clippy suggested this, but it seems it's just the generic `branches_sharing_code` lint that fires, and it's quite subtle to refactor correctly in this case.

Reply:
I had a go at restoring the previous functionality while dodging Clippy in f19be0de21c1ca52f85fabede1b685c78f0e2138 (seeing as Sean is afk).

Reply:
me approves

Reply:
tho I wouldn't mind changing this to use something different to equality. Hinders readability and doesn't really feel quite right in hindsight. But this is out of scope for the current PR. I'm fine with @michaelsproul 's solution :)

Reply:
Yeah, I agree it should be a separate method like `same_status`, `same_kind`, or something

Reply:
thanks guys 🙂